docs/pages/reference/react/types.mdx

```diff
@@ -99,4 +99,10 @@ export type FrameButtonPostRedirectProvidedProps = {
   /** a label to display on the button */
   children: string | number;
 };
+
+/** See https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional */
+export type NextJSServerPageProps = {
```

Comment:
Perhaps rename to `NextServerPageProps`? More in line with types such as `NextRequest`